turbopack/crates/turbo-rcstr/src/dynamic.rs

```diff
@@ -0,0 +1,76 @@
+use std::{
+    borrow::Cow,
+    hash::{Hash, Hasher},
+    ptr::NonNull,
+};
+
+use triomphe::Arc;
+
+use crate::{
+    tagged_value::{TaggedValue, MAX_INLINE_LEN},
+    RcStr, INLINE_TAG_INIT, LEN_OFFSET, TAG_MASK,
+};
+
+#[derive(Debug)]
+pub(crate) struct Entry {
+    pub string: Arc<str>,
```

Comment:
I'm not opposed to changing this if benchmarks show an improvement, but should this still be `Arc<String>`? I thought the idea of the original version of `RcStr` is that you wanted really fast conversions to/from `String`?

Sidenote: I'm fine with owned `String` conversions to/from interned strings being more expensive, because they're bounded in length, I think we were mostly concerned about O(n^2) blowups from long string concatenation operations in a loop.